src/components/Card.js

```diff
@@ -1,50 +1,74 @@
-class Card {
-  constructor(data, cardSelector, _handleImageClick) {
-    this._name = data.name;
-    this._link = data.link;
+export default class Card {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleCardClick,
+    handleTrashClick
+    // handleDeleteConfirm
+    // cardId,
+    // userId
+  ) {
+    this._name = name;
+    this._link = link;
     this._cardSelector = cardSelector;
-    this._handleImageClick = _handleImageClick;
+    this._handleCardClick = handleCardClick;
+    this._handleTrashClick = handleTrashClick;
+    // this._handleDeleteConfirm = handleDeleteConfirm;
+    this._cardId = _id;
+    this._isLiked = isLiked;
+    // this._userId = userId;
   }
 
   _setEventListeners() {
-    this._likeButton.addEventListener("click", () => this._handleLikeIcon());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteIcon()
+    // ".card__like-button
+
+    this._likeButton = this._cardElement.querySelector(".card__like-button");
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeButton();
+    });
+
+    this._cardImageEl.addEventListener("click", () => {
+      this._handleCardClick({ name: this._name, link: this._link });
+    });
+
+    this._cardTrashButton = this._cardElement.querySelector(
+      ".card__trash-button"
     );
-    this._cardImage.addEventListener("click", () => {
-      this._handleImageClick(this);
+    this._cardTrashButton.addEventListener("click", (inputValues) => {
+      console.log("card Trash Button clicked");
+      this._handleTrashClick(inputValues);
+      console.log("card ID is ", this._cardId);
+      // this._handleDeleteConfirm(this._cardId);
+      // this._handleDeleteClick(this._cardId, this._userId);
+      // this._handleTrashButton();
     });
   }
 
-  _handleLikeIcon() {
+  _handleLikeButton() {
     this._likeButton.classList.toggle("card__like-button_active");
```

Comment:
Needs correction: You should update the likes on the server and change the button state only after that

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```